setup.py

```diff
@@ -1,81 +1,68 @@
+from flit_core import common, config
+from setuptools_scm.integration import find_files
 
-try:
-    import pytoml
-except ImportError:
-    sys.exit('Please use `pip install .` or install pytoml first.')
 
-proj = pytoml.loads(Path('pyproject.toml').read_text())
-metadata = proj['tool']['scanpy']
-
-setup(
-    name='scanpy',
-    use_scm_version=True,
-    setup_requires=['setuptools_scm'],
-    description='Single-Cell Analysis in Python.',
-    long_description=Path('README.rst').read_text('utf-8'),
-    url='http://github.com/theislab/scanpy',
-    author=metadata['author'],
-    author_email=metadata['author-email'],
-    license='BSD',
-    python_requires='>=3.6',
-    install_requires=[
-        l.strip() for l in Path('requirements.txt').read_text('utf-8').splitlines()
-    ],
-    extras_require=dict(
-        louvain=['python-igraph', 'louvain>=0.6,!=0.6.2'],
-        leiden=['python-igraph', 'leidenalg'],
-        bbknn=['bbknn'],
-        scvi=['scvi==0.6.7'],
-        rapids=['cudf>=0.9', 'cuml>=0.9', 'cugraph>=0.9'],
-        magic=['magic-impute>=2.0'],
-        skmisc=['scikit-misc>=0.1.3'],
-        harmony=['harmonypy'],
-        scanorama=['scanorama'],
-        scrublet=['scrublet'],
-        dev=['setuptools_scm', 'pytoml', 'black>=20.8b1'],
-        doc=[
-            'sphinx>=3.2',
-            'sphinx_rtd_theme>=0.3.1',
-            'sphinx_autodoc_typehints',
-            'scanpydoc>=0.5',
-            'typing_extensions; python_version < "3.8"',  # for `Literal`
-        ],
-        test=[
-            'pytest>=4.4',
-            'pytest-nunit',
-            'dask[array]!=2.17.0',
-            'fsspec',
-            'zappy',
-            'zarr',
-            'profimp',
-        ],
-    ),
-    packages=find_packages(),
-    include_package_data=True,
-    entry_points=dict(console_scripts=['scanpy=scanpy.cli:console_main']),
-    zip_safe=False,
-    classifiers=[
-        'Development Status :: 5 - Production/Stable',
-        'Environment :: Console',
-        'Framework :: Jupyter',
-        'Intended Audience :: Developers',
-        'Intended Audience :: Science/Research',
-        'Natural Language :: English',
-        'Operating System :: MacOS :: MacOS X',
-        'Operating System :: Microsoft :: Windows',
-        'Operating System :: POSIX :: Linux',
-        'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.5',
-        'Programming Language :: Python :: 3.6',
-        'Programming Language :: Python :: 3.7',
-        'Topic :: Scientific/Engineering :: Bio-Informatics',
-        'Topic :: Scientific/Engineering :: Visualization',
-    ],
+field_names = dict(
+    {n: n for n in "name version author author_email license classifiers".split()},
+    description="summary",
+    long_description="description",
+    long_description_content_type="description_content_type",
+    python_requires="requires_python",
+    url="home_page",
 )
```

Comment:
A bit cleaner/ more consistent:

```python
field_names = {
      "description": "summary",
      "long_description": "description",
      "long_description_content_type": "description_content_type",
      "python_requires": "requires_python",
      "url": "home_page",
      **{n: n for n in ["name", "version", "author", "author_email", "license", "classifiers"]},
}
```

Reply:
Dicts that contain keyword arguments should be created using the `dict` constructor, otherwise I agree. :+1: